src/test/java/org/fogbowcloud/manager/requests/api/local/http/ImageRequestControllerTest.java

```diff
@@ -0,0 +1,164 @@
+        PowerMockito.mockStatic(ApplicationFacade.class);
+        given(ApplicationFacade.getInstance()).willReturn(this.facade);
+    }
+
+    // test case: Test getAllImages() when no images were found, it must return an empty JSON.
+    @Test
+    public void testGetAllImagesWhenHasNoData() throws Exception {
+        // set up
+        Map<String, String> imagesMap = new HashMap<>();
+        doReturn(imagesMap).when(this.facade).getAllImages(anyString(), anyString());
+
+        RequestBuilder requestBuilder = createRequestBuilder(IMAGE_ENDPOINT, getHttpHeaders(), "");
+
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        assertEquals("{}", result.getResponse().getContentAsString());
+        assertEquals(expectedStatus, result.getResponse().getStatus());
+    }
+
+    // test case: Test getAllImages() when there are some images.
+    @Test
+    public void testGetAllImagesWhenHasData() throws Exception {
+        // set up
+        Map<String, String> imagesMap = new HashMap<>();
+        imagesMap.put("image-id1", "image-name1");
+        imagesMap.put("image-id2", "image-name2");
+        imagesMap.put("image-id3", "image-name3");
+
+        doReturn(imagesMap).when(this.facade).getAllImages(anyString(), anyString());
+
+        RequestBuilder requestBuilder = createRequestBuilder(IMAGE_ENDPOINT, getHttpHeaders(), "");
+
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        assertEquals(expectedStatus, result.getResponse().getStatus());
+
+        TypeToken<Map<String, String>> token = new TypeToken<Map<String, String>>(){};
+        Map<String, String> resultMap = new Gson().fromJson(result.getResponse().getContentAsString(), token.getType());
+        assertTrue(resultMap.size() == 3);
+    }
+
+    // test case: Test if given an existing image id, the getImage() returns that image properly.
+    @Test
+    public void testGetImageById() throws Exception {
+        // set up
+        String fakeId = "fake-Id-1";
+        String imageEndpoint = IMAGE_ENDPOINT + "/" + fakeId;
+
+        Image image = new Image(fakeId, "fake-name", 1, 1, 1, "READY");
+
+        doReturn(image).when(this.facade).getImage(anyString(), anyString(), anyString());
+
+        RequestBuilder requestBuilder = createRequestBuilder(imageEndpoint, getHttpHeaders(), "");
+
+        // exercise
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        assertEquals(expectedStatus, result.getResponse().getStatus());
+
+        Image resultImage = new Gson().fromJson(result.getResponse().getContentAsString(), Image.class);
+        assertEquals(resultImage.getId(), image.getId());
```

Comment:
@llvieira I think you swapped the args. Remember to add first the expected and then the actual. Please, verify the other test cases accordingly.

Reply:
Done!